### ISSUE TITLE
Looting 1/2/3 Upgrades Reducing drop chance/fluxing chance on vanilla mobs

### ISSUE THREAD

**Issue by Hachuurui** [Ipsis/Woot#371]
I've noticed a few other issues related to these upgrades. 

Expected: Drop rate on items increases with each looting upgrade.

What happens: Drop rate becomes erratic and normally stays on the lower end of the percentage spectrum. T3 upgrade is always the lowest rate and does not change.

I've taken a moment to collect some data and information.

![2018-07-23_17 41 12](https://user-images.githubusercontent.com/2464584/43104914-07c5b96e-8ea1-11e8-99ef-6c5f1bf797eb.png)
![2018-07-23_17 42 03](https://user-images.githubusercontent.com/2464584/43104915-07ccfcba-8ea1-11e8-8268-b8c635adf58d.png)
![2018-07-23_17 42 05](https://user-images.githubusercontent.com/2464584/43104916-07d58c86-8ea1-11e8-9745-d59519b5d803.png)
![2018-07-23_17 42 47](https://user-images.githubusercontent.com/2464584/43104917-07dfb742-8ea1-11e8-85f9-f52de619d247.png)
![2018-07-23_17 43 09](https://user-images.githubusercontent.com/2464584/43104918-07e866bc-8ea1-11e8-99e0-863b9a7e9568.png)
![2018-07-23_17 43 24](https://user-images.githubusercontent.com/2464584/43104919-07f0e1de-8ea1-11e8-8256-3e18362c6e65.png)

The base rate never changes. The T3 Looting also never changes and is always the lowest drop rate.

The images above show the overall reduction and fluxing changes depending on Tier.

The list of mods in this pack is as follows:

Actually Additions
Advanced Rocketry
Apple Skin
Applied Energistics 2
Architecture Craft
Armory Expansion
Aroma1997 Core
Aroma 1997 Dimensional World
Astral Sorcery
Auto Reg Lib
Bad Wither No Cookie - Reloaded
Baubles
Bd Lib
Better Builder's Wands
Better FPS
BiblioCraft
Binnie's Mods
Biomes O'Plenty
Blood Magic
Bonsai Trees
Bookshelf
Botania
Brandon's Core
Career Bees
Chameleon
Chisel
Chisel & Bits
Code Chicken Lib
CoFH Core
CoFH World
Compact Machines
Connected Textures Mod
Construct's Armory
Controlling
Cooking for blockheads
Cosmetic Armor Reworked
Craft Tweaker
Creative Core
Cucumber Libray
Dark Utilities
Draconic Evolution
Dynamic Surroundings
Embers
Ender IO
Ender Storage
Ender Core
Environmental Tech
Exchangers
Extra Utilities 2
Extra Cells 2
Extreme Reactors
Fairy Lights
Farseek
Flat Colored Blocks
Forestry
FTB Utilities
FTB Lib
Guide API
Hats
Hunting Dimension
iChunUtil
Immersive Engineering
Immersive Petroleum
Industrial Foregoing
Inventory Tweaks
Iron Chests
Item Scroller
JAOPCA
JEI Bees
Journey Map
Just a Few Fish
Just Enough Items
Just Enough IDS
Lib Vulpes
Little Tiles
Magic Bees
Malisis Core
Malisis Doors
Mantle
McJtyLib
MC MultiPart
Mekanism
Mekanism Generators
Mob Grinding Utils
Mod Name Tooltip
Mod Tweaker
Morpheus
MrCrayfish's Furniture Mod
MT Lib
Mystical agradditions
Mystical Agriculture
Nature's Compass
No Recipe Book
Not Enough Wands
Open Blocks
Open Computers
Open Mods Lib
Ore Excavation
Ore Excavation Integration
p455w0rd's Library
Pam's HarvestCraft
Pressure Pipes
Psi
Quantum Storage
Quark
Rail Roads
ReAuth
Reborn Core
Reborn Storage
Redstone Flux
Refined Storage
RF Tools
RF Tools control
RF Tools Dimensions
Roguelike Dungeons
Shadowfacts' Foreglin
Snad
Soot
Stimmedcow : No More Recipe Conflict
Storage Drawers
Streams
Tall Gates
Tesla Core Lib
Thaumcraft
Thaumic Inventory Scanning
Thaummic JEI
The One Probe
The Weirding Gadget
Thermal Cultivation
Thermal Dynamics
Thermal Expansion
Thermal Foundation
Thermal Innovation
Tinkers Construct
Tinkers' Tool Leveling
Tomb Many Graves 2
Tomb Many Graves 2 API
Tomb Many Plugins - Cosmetic Armor Reworked
TOP Addons
Torch Master
Trumpet Skeleton
Valkyrie Lib
Vanilla Fix
ViesCraft - Airships
Waystones
Wireless Crafting Terminal
Woot
Wrap UP
XNet
YABBA
ZeroCore

**Comment by Hachuurui** [comment]
I will also note, our servers do not use sponge.

The modpack in use can be found here: https://minecraft.curseforge.com/projects/the-ryrypatpatpackpack

**Comment by Ipsis** [comment]
Sorry about the delay in replying. Dealing with some crash bugs.

So this is an intriguing one, so as a start I'll explain how Woot works out the drop rate and see if that suggest anything. 

Woot will kill 500 mobs of your chosen type at the specific looting level in a separate dimension and learn from that the percentage drops of each item. Once it has learned from all 500 kills, then that mob/looting level is fixed and stored when you end the game. If you change looting level then it will pause learning the current level and start learning the new level.
While it is learning the drop rate will fluctuate as new information comes in.

It takes about 500 seconds to process all the kills for a single looting level. (You can see how many kills is has learned from with the command "/woot dump learning".)

So I ran with a creeper at all looting levels and let learning complete and I get the following results.
No looting - 38%
Looting I - 34%
Looting II -  36%
Looting III -  30%

I think I know what is going on and I can see why it looks wrong to you.

Basically the looting level changes the stack size of the drop but not the chance of getting the drop.
So you will always have roughly a ~35% change of gunpowder, but with a higher looting level you have a better chance of that gunpowder drop being a larger stack.

When Woot picks the drop it will decide if gunpowder should be generated, then it looks at a second list to see the percentage chances of each stack size. However that isn't shown on the console, just the top level item chance. (I can maybe work that into the tooltip of the item to show the stack size chances).

So if you stick a chest on the output of the factory, you should end up with more gunpowder from the one with the higher looting level.

**Comment by Hachuurui** [comment]
Thanks for replying, and no need to apologize as this really isn't something that I would consider a critical bug as a developer.

This is a bit interesting though. I had it running for about week at level 3 and saw it set at 28% drop rate but did not consider the stack concept.  I am not sure if I'd be able to run that command as a user but I can just update the test environment we used to build the server and continue research there. Knowing that they stabilize after a while eliminates my concerns, as I know that we can balance rates using the configuration files you provide.  I'll close this and continue testing on my side to look into upping rates using the custom drop rate config. 

Thank you for your time and for the explaination.
